doc/classes/Rect2i.xml

```diff
@@ -114,79 +136,93 @@
-				Returns a copy of the [Rect2i] grown by the specified [param amount] on the specified [enum Side].
+				Returns a copy of this rectangle with its [param side] extended by the given [param amount] (see [enum Side] constants). A negative [param amount] shrinks the rectangle, instead. See also [method grow] and [method grow_individual].
 			</description>
 		</method>
 		<method name="has_area" qualifiers="const">
 			<return type="bool" />
 			<description>
-				Returns [code]true[/code] if the [Rect2i] has area, and [code]false[/code] if the [Rect2i] is linear, empty, or has a negative [member size]. See also [method get_area].
+				Returns [code]true[/code] if the rectangle has area, or [code]false[/code] if the rectangle is linear, empty, or the [member size] is negative. See also [method get_area].
 			</description>
 		</method>
 		<method name="has_point" qualifiers="const">
 			<return type="bool" />
 			<param index="0" name="point" type="Vector2i" />
 			<description>
-				Returns [code]true[/code] if the [Rect2i] contains a point. By convention, the right and bottom edges of the [Rect2i] are considered exclusive, so points on these edges are [b]not[/b] included.
-				[b]Note:[/b] This method is not reliable for [Rect2i] with a [i]negative size[/i]. Use [method abs] to get a positive sized equivalent rectangle to check for contained points.
+				Returns [code]true[/code] if the rectangle contains the given [param point]. By convention, points on the right and bottom edges are [b]not[/b] included.
+				[b]Note:[/b] This method is not reliable for [Rect2] with a [i]negative[/i] [member size]. Use [method abs] first to get a valid rectangle.
 			</description>
 		</method>
 		<method name="intersection" qualifiers="const">
 			<return type="Rect2i" />
 			<param index="0" name="b" type="Rect2i" />
 			<description>
-				Returns the intersection of this [Rect2i] and [code]b[/code].
-				If the rectangles do not intersect, an empty [Rect2i] is returned.
+				Returns the intersection between this rectangle and [param b]. If the rectangles do not intersect, returns an empty [Rect2].
+				[codeblocks]
+				[gdscript]
+				var a = Rect2i(0, 0, 5, 10)
+				var b = Rect2i(2, 0, 8, 4)
+
+				var c = a.intersection(b) # c is Rect2i(2, 0, 3, 4)
+				[/gdscript]
+				[csharp]
+				var a = new Rect2I(0, 0, 5, 10);
+				var b = new Rect2I(2, 0, 8, 4);
+
+				var c = rect1.Intersection(rect2); // c is Rect2I(2, 0, 3, 4)
+				[/csharp]
+				[/codeblocks]
+				[b]Note:[/b] If you only need to know whether two rectangles are overlapping, use [method intersects], instead.
 			</description>
 		</method>
 		<method name="intersects" qualifiers="const">
 			<return type="bool" />
 			<param index="0" name="b" type="Rect2i" />
 			<description>
-				Returns [code]true[/code] if the [Rect2i] overlaps with [code]b[/code] (i.e. they have at least one point in common).
+				Returns [code]true[/code] if this rectangle overlaps with the [param b] rectangle.
```

Comment:
The explainer is useful. Sharing a corner or an edge counts as intersecting, ri-i-ight.

Reply:
It does, it does.

Reply:
Corners and edges do not count. It sounds weird but it is what it is. Unlike the **Rect2** equivalent, the **Rect2i** `intersects` does not have the additional `include_borders` argument. Because it may be confusing, it is proooobably worth mentioning, too.